doc/source/indexing.rst

```diff
@@ -292,6 +292,28 @@ Selection By Label
    This is sometimes called ``chained assignment`` and should be avoided.
    See :ref:`Returning a View versus Copy <indexing.view_versus_copy>`
 
+.. warning::
+
+   ``.loc`` is strict when you present slicers that are not compatible with the index type. For example
+   using integers in a ``DatetimeIndex`` or float indexers in an ``Int64Index``. These will raise a ``TypeError``
```

Comment:
Only with a small exception on this rule that we try to parse strings as datetimes for a DatetimeIndex (but maybe not needed to say this here)